logstash-core/src/main/java/org/logstash/execution/Filter.java

```diff
@@ -0,0 +1,68 @@
+package org.logstash.execution;
+
+import org.logstash.Event;
+
+/**
+ * A Filter is simply a mapping of {@link QueueReader} to a new {@link QueueReader}.
+ */
+public interface Filter extends AutoCloseable {
+
+    QueueReader filter(QueueReader reader);
+
+    void flush(boolean isShutdown);
+
+    @LogstashPlugin(name = "mutate")
+    final class Mutate implements Filter {
+
+        private final String field;
+
+        private final String value;
+
+        /**
+         * Required Constructor Signature only taking a {@link LsConfiguration}.
+         * @param configuration Logstash Configuration
+         */
+        public Mutate(final LsConfiguration configuration) {
+            this.field = configuration.getString("ls.plugin.mutate.field");
```

Comment:
How does core know what is valid / not valid configuration and type ? 

For example, i configure:
```
  mutate {
       junk => { "foo" => "bar" }
   }
```
How does Logstash know to error ?

Does this require an external file to describe the valid configuration ? Also, where would deprecation warnings, obsoleted warnings, and validation happen ?  

If it is an external file, can you provide an example of what that may look like ?

Reply:
adding another comment, since GH thinks this is outdated by a commit.